### PR TITLE
8328827: Convert java/awt/print/PrinterJob/PrinterDialogsModalityTest/PrinterDialogsModalityTest.html applet test to main

### PR DESCRIPTION
**Pull request by DamonGuy**
Convert java/awt/print/PrinterJob/PrinterDialogsModalityTest/PrinterDialogsModalityTest.html applet test to main using PassFailJFrame

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328827](https://bugs.openjdk.org/browse/JDK-8328827): Convert java/awt/print/PrinterJob/PrinterDialogsModalityTest/PrinterDialogsModalityTest.html applet test to main (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [c39292d0](https://git.openjdk.org/jdk/pull/18461/files/c39292d0f1da0100950fbfc48e0bf633b6170580)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer) ⚠️ Review applies to [d61f55fe](https://git.openjdk.org/jdk/pull/18461/files/d61f55fe77af416c43ec27c7800b2891eb0fa285)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18461/head:pull/18461` \
`$ git checkout pull/18461`

Update a local copy of the PR: \
`$ git checkout pull/18461` \
`$ git pull https://git.openjdk.org/jdk.git pull/18461/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18461`

View PR using the GUI difftool: \
`$ git pr show -t 18461`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18461.diff">https://git.openjdk.org/jdk/pull/18461.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18461#issuecomment-2015652218)

### CODE REVIEW

File: test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import javax.swing.JFrame;
+
+public class PrinterDialogsModalityTest {
+    private static final String INSTRUCTIONS =
+            """
+             After the test starts, you will see a frame titled "Test Frame"
+             with two buttons: "Page Dialog" and "Print Dialog".
+             1. Make the "Test Frame" active by clicking on title.
+             2. Press "Page Dialog" button and a page dailog should popup.
```
Comment (prrace) [inline test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java:49]: dailog -> dialog

File: test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+             If you are able to execute all steps successfully then the test
+             passes, otherwise it fails.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame
+                .builder()
+                .title("PrinterDialogsModalityTest Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
```
Comment (prrace) [inline test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java:68]: Oh, that's a nice idea. I'd been tweaking by hand.

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java (hunk 1 above)
Comment (DamonGuy) [reply test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java]: Fixed. Thanks.

File: test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java (hunk 2 above)
Comment (DamonGuy) [reply test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java:68]: I agree it's a nice idea, but I took this idea from other PR's I've reviewed. Not sure who started this first, but very handy.

**Comment by honkar-jdk** [comment]
@DamonGuy The test in https://github.com/openjdk/jdk/pull/18362 and the one in this PR test similar functionality -Dialog Modality when using Print functions. Will combining the tests make more sense instead of 2 separate tests?

File: test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        });
+        Button print = new Button("Print Dialog");
+        print.addActionListener(e -> {
+            PrinterJob prnJob = PrinterJob.getPrinterJob();
+            prnJob.printDialog();
+        });
+        frame.add(page);
+        frame.add(print);
+        frame.pack();
+        frame.setVisible(true);
```
Comment (honkar-jdk) [inline test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java:92]: frame.setVisible(true); not required when using PassFailJFrame.

Comment (DamonGuy) [reply test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java]: Good point. Fixed. Thanks

**Comment by DamonGuy** [comment]
> @DamonGuy The test in #18362 and the one in this PR test similar functionality -Dialog Modality when using Print functions. Will combining or retaining only one test make more sense instead of 2 separate tests?

Good point. Not sure. This one was originally for windows only for print dialog and page dialog modality. The other PR was for web dialogs. They are similar though now. What do you suggest @honkar-jdk? I prefer this test overall over the other if I had to pick one. Cleaner and more succinct test.

**Comment by honkar-jdk** [comment]
> > @DamonGuy The test in #18362 and the one in this PR test similar functionality -Dialog Modality when using Print functions. Will combining or retaining only one test make more sense instead of 2 separate tests?
> 
> Good point. This one was originally for windows only for print dialog and page dialog modality. The other PR was for web dialogs. They're also in different directories, one for `PrinterJob` and one for `Dialog`. They are similar though now. What do you suggest @honkar-jdk? I prefer this test overall over the other if I had to pick one. Cleaner and more succinct test.

In my opinion PrintModalDialog.java can be removed since it is redundant. I agree this test looks better. If we go ahead with removing PrintModalDialog.java then

- Remove PrintModalDialog.java as part of new PR.
- This test (PrinterDialogsModalityTest.java) can be extended to all platforms and we no longer need @requires tag. 
- Probably update the @bug to include the bug id related to PrintModalDialog.java.

If it is not too much effort then I suggest on deleting the redundant test. @prrace Any suggestions?

File: test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import java.awt.print.PageFormat;
+import java.awt.print.PrinterJob;
+
+import javax.swing.JFrame;
+
+/*
+ * @test
+ * @bug 4784285 4785920
+ * @key printer
+ * @requires (os.family == "windows")
```
Comment (honkar-jdk) [inline test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java:35]: Even if we retain both the tests, I believe this test can be extended to all platforms and `@requires` tag may not be required.

Comment (DamonGuy) [reply test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java]: Removed.

File: test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                .title("PrinterDialogsModalityTest Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(PrinterDialogsModalityTest::init)
+                .build()
+                .awaitAndCheck();
+    }
+
+    public static JFrame init() {
```
Comment (TejeshR13) [inline test/jdk/java/awt/print/PrinterJob/PrinterDialogsModalityTest.java:74]: Can rename the `init` method.

**Review by TejeshR13 — APPROVED** [review]

**Comment by prrace** [comment]
> * we need to remove PrintModalDialog.java as part of new PR since it is already integrated.
> * This test (PrinterDialogsModalityTest.java) can be extended to all platforms and we no longer need `@requires` tag.
> * Probably update the `@bug` to include the bug id related to PrintModalDialog.java.
> 
> If it is not too much effort then I suggest on deleting the redundant test. @prrace Any suggestions?

Yes, this sounds fine to me.

**Pull request closed without merging**